dask_expr/_shuffle.py

```diff
@@ -716,6 +734,80 @@ def _simplify_up(self, parent):
             )
 
 
+class SortValues(BaseSetIndexSortValues):
+    _parameters = [
+        "frame",
+        "by",
+        "ascending",
+        "na_position",
+        "npartitions",
+        "partition_size",
+        "sort_function",
+        "sort_function_kwargs",
+    ]
+    _defaults = {
+        "partition_size": 128e6,
+        "ascending": True,
+        "npartitions": None,
+        "na_position": "last",
+        "sort_function": None,
+        "sort_function_kwargs": None,
+    }
+
+    @property
+    def sort_function(self):
+        if self.operand("sort_function") is not None:
+            return self.operand("sort_function")
+        return M.sort_values
+
+    @property
+    def sort_function_kwargs(self):
+        sort_kwargs = {
+            "by": self.by,
+            "ascending": self.ascending,
+            "na_position": self.na_position,
+        }
+        if self.operand("sort_function_kwargs") is not None:
+            sort_kwargs.update(self.operand("sort_function_kwargs"))
+        return sort_kwargs
+
+    @property
+    def _meta(self):
+        return self.frame._meta
+
+    def _lower(self):
+        by = self.frame[self.by[0]]
+        divisions, _, _, presorted = _calculate_divisions(
```

Comment:
As noted above: We may have already executed this calculation earlier (if we happend to query `divisions`), so it would be nice to avoid another `compute` call here.

Reply:
Cached, see above